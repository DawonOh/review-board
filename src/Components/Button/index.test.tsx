import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import 'jest-styled-components';
import { Button } from '.';

test('renders Button component correctly', () => {
  const { container } = render(<Button content="Button Test" />);
  const content = screen.getByText('Button Test');
  expect(content).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

test('renders Container Element correctly', () => {
  render(<Button content="Button Test" />);
  const container = screen.getByTestId('container-element');
  expect(container).toHaveStyleRule('background-color', '#E0E0E0');

  const content = screen.getByText('Button Test');
  expect(content).toHaveStyleRule('color', '#000');
});

test('Button component has backgroundColor,color and size props', () => {
  render(
    <Button
      content="Button Test"
      backgroundColor="#FF5959"
      color="#fff"
      size="0.3em"
    />
  );
  const container = screen.getByTestId('container-element');
  expect(container).toHaveStyleRule('background-color', '#FF5959');
  expect(container).toHaveStyleRule('padding', '0.3em');

  const content = screen.getByText('Button Test');
  expect(content).toHaveStyleRule('color', '#fff');
});

test('click the button', () => {
  const handleClick = jest.fn();
  render(<Button content="Button Test" onClick={handleClick} />);

  const content = screen.getByText('Button Test');
  expect(handleClick).toHaveBeenCalledTimes(0);
  fireEvent.click(content);
  expect(handleClick).toHaveBeenCalledTimes(1);
});
