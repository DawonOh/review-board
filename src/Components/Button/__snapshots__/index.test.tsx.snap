// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button component correctly 1`] = `
.c0 {
  padding: 0.6em;
  background-color: #E0E0E0;
  border: none;
  border-radius: 0.3em;
  cursor: pointer;
}

.c1 {
  color: #000;
}

<div>
  <button
    class="c0"
    color="#000"
    data-testid="container-element"
  >
    <span
      class="c1"
      color="#000"
    >
      Button Test
    </span>
  </button>
</div>
`;
